app/decorators/decidim/admin/newsletters_controller_decorator.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+require_dependency 'decidim/admin/newsletters_controller'
+
+# Sort Admins by role and area
+::Decidim::Admin::NewslettersController.class_eval do
+  alias_method :original_collection, :collection
+
+  private
+
+  def collection
+    return original_collection unless current_user.department_admin?
+    
+    @collection ||= Decidim::Newsletter.where(organization: current_organization)
+                                       .joins(author: :areas)
+                                       .where("department_admin_areas.decidim_area_id = ?", current_user_areas.pluck(:id))
+
+  end
+
+  def current_user_areas
+    return unless current_user.department_admin?
+    return unless current_user.areas.any?  
+    return current_user.areas if current_user&.department_admin?
```

Comment:
Shouldn't this line be enough for this method?
Can't we remove the previous two lines?

Reply:
You are right! Change applied!

Reply:
I like more the one liner, but well...